### PR TITLE
align "access_secret_key" to "secret_access_key"

### PR DESCRIPTION
**Pull request by johnnyaug**
Solves #501

### CODE REVIEW

**Review by arielshaqed — COMMENTED** [review]
Oh yeah!

The thing is, this breaks every single config file out there.  You're not going to believe this, I'm a
# :flags:  Bearer of Good News :unicorn: 
here...

In #1729 we're switching to deeper use of viper, mostly to get validation.  *But* I discovered that viper also has a way to alias one key onto the other -- "[Registering and Using Aliases](https://pkg.go.dev/github.com/spf13/viper#readme-registering-and-using-aliases)".  So we can keep the old name and just stop documenting it.  (And of course add that to config_test)

WDYT?

**Comment by arielshaqed** [comment]
May fix #501.

File: pkg/config/config.go [hunk 1]
```diff
@@ -99,12 +99,15 @@ const (
...
-	BlockstoreS3MaxRetriesKey            = "blockstore.s3.max_retries"
+	BlockstoreTypeKey                         = "blockstore.type"
+	BlockstoreLocalPathKey                    = "blockstore.local.path"
+	BlockstoreS3RegionKey                     = "blockstore.s3.region"
+	BlockstoreS3StreamingChunkSizeKey         = "blockstore.s3.streaming_chunk_size"
+	BlockstoreS3StreamingChunkTimeoutKey      = "blockstore.s3.streaming_chunk_timeout"
+	BlockstoreS3MaxRetriesKey                 = "blockstore.s3.max_retries"
+	BlockstoreS3CredentialsAccessKeyIdKey     = "blockstore.s3.credentials.access_key_id"
+	BlockstoreS3CredentialsSecretAccessKeyKey = "blockstore.s3.credentials.secret_access_key"
+	BlockstoreS3CredentialsSessionTokenKey    = "blockstore.s3.credentials.session_token"
```
Comment (arielshaqed) [inline pkg/config/config.go:110]: I'm in the middle of a mega-cleanup of config.go, #1729.  These string keys will soon be gone (all of them, entirely), so please avoid fixing them for now.  I promise I shall eradicate them.
Sorry for not being specific about this when I said I was "doing something something".

File: pkg/config/config.go [hunk 2]
```diff
@@ -174,6 +177,8 @@ func setDefaults() {
 
 	viper.SetDefault(BlockstoreAzureTryTimeoutKey, DefaultAzureTryTimeout)
 	viper.SetDefault(BlockstoreAzureAuthMethod, DefaultAzureAuthMethod)
+	// allow using deprecated key for AWS secret key:
+	viper.RegisterAlias(BlockstoreS3CredentialsSecretAccessKeyKey, "blockstore.s3.credentials.access_secret_key")
```
Comment (arielshaqed) [inline pkg/config/config.go:181]:
```suggestion
	viper.RegisterAlias("blockstore.s3.credentials.access_secret_key", BlockstoreS3CredentialsSecretAccessKeyKey)
```
I think the old name should be the alias, the new name should be the new key.

**Review by arielshaqed — COMMENTED** [review]
Great, thanks!

Please _don't_ give names to any other keys, I am eliminating all of these string names and it will only make my conflicts worse.

Approving as soon as those named `*Key`s are gone.

File: pkg/config/config.go [hunk 3]
```diff
@@ -174,6 +174,8 @@ func setDefaults() {
 
 	viper.SetDefault(BlockstoreAzureTryTimeoutKey, DefaultAzureTryTimeout)
 	viper.SetDefault(BlockstoreAzureAuthMethod, DefaultAzureAuthMethod)
+	// allow using deprecated key for AWS secret key:
+	viper.RegisterAlias("blockstore.s3.credentials.secret_access_key", "blockstore.s3.credentials.access_secret_key")
```
Comment (arielshaqed) [inline pkg/config/config.go:178]: Let's test this alias in at least one place. You know,  for BC.

**Review by arielshaqed — APPROVED** [review]
Very cool, thanks! We might want to test that the old version still works,  say in auth_test.

**Review by arielshaqed — APPROVED** [review]
Thanks!  The no-aliasing hack is unavoidable but truly horrible.  Can you add a comment to this PR to link to the Viper bug?

**Comment by johnnyaug** [comment]
This is a workaround to https://github.com/spf13/viper/issues/689

**Comment by arielshaqed** [comment]
> This is a workaround to bypass [spf13/viper#689](https://github.com/spf13/viper/issues/689)

That bug has a PR to fix it. And that PR will be 3 tomorrow!

@johnnyaug @nopcoder istio have [a fork of viper](https://github.com/istio/istio/pull/13594/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R9), which may have fixed it (the istio PR links to 689). Perhaps time to investigate whether the istio viper is better for us than the spf13 viper?

**Comment by nopcoder** [comment]
Thanks @arielshaqed, will take a look tomorrow the difference.

**Pull request merged**